src/Text/Pandoc/Class/PandocMonad.hs

```diff
@@ -638,6 +652,7 @@ makeCanonical :: FilePath -> FilePath
 makeCanonical = Posix.joinPath . transformPathParts . splitDirectories
  where  transformPathParts = reverse . foldl' go []
         go as     "."  = as
+        go ("..":as) ".." = ["..", ".."] <> as
```

Comment:
Maybe this should be in a separate PR?

Reply:
Is this really a bug?  Seems to me that this case will never occur.
We start with the accumulator as an empty list.
The only way something can get added to that list is if it matches the third clause `go as x`.  But we will never match that with `x` == `".."`, because `".."` always matches the second clause and gets discarded.  Am I missing something?

Reply:
Without line 655, `makeCanonical "../../file"` will evaluate to `"file"`. I think it goes like this:

1. `go [] ".." = ".." : []` (657)
2. `go [_:[]] ".." = []` (656)
3. `go [] "file" = ["file"]` (657)

With line 655:

1. `go [] ".." = ".." : []` (657)
2. `go ("..":[]) ".." = ["..", ".."] <> []` (655)
3. `go ["..", ".."] "file" = "file" : ["..", ".."]` (657)

I see that I messed up the alignment, will fix.

Reply:
Oh right!  We don't match that second line when `as` is empty.

Reply:
I'd prefer if this were  a separate commit (not necessarily in a separate PR), so it stands out in the history.

Reply:
Done!